### PR TITLE
Feature/5/TINY-8595: UI components

### PR DESCRIPTION
**Pull request by hamza0867**
Related Ticket: TINY-8595

Description of Changes:
This PR updates the docs of the UI components section.
Worth noting that there is one xref that doesn't resolve but should get fixed automatically once #2328 gets merged

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/pages/ui-components/dialogcomponents.adoc [hunk 1]
```diff
@@ -568,8 +577,8 @@ A *textarea* is a multiline text field.
 ==== urlinput
 
 A *urlinput* is a specialized composite component for URL input or file upload. It has a label, a text input field and an optional filepicker button. The urlinput component also includes a *typeahead* dropdown that will display previously-entered URLs that match the current input text and update as the user types.
-
-NOTE: The filepicker button will only appear if link:{baseurl}/configure/file-image-upload/#file_picker_callback[`file_picker_callback`] is configured.
+// TODO: the xref in the next line should be fixed automatically once #2328 gets merged
```
[the remark this answers is not in the document]
Comment (hamza0867) [reply modules/ROOT/pages/ui-components/dialogcomponents.adoc]: Done

File: modules/ROOT/pages/ui-components/menuitems.adoc [hunk 2]
```diff
@@ -44,7 +44,7 @@ tinymce.init({
 });
 ----
```
[the remark this answers is not in the document]
Comment (hamza0867) [reply modules/ROOT/pages/ui-components/menuitems.adoc:46]: Done

File: modules/ROOT/pages/ui-components/contextmenu.adoc [hunk 3]
```diff
@@ -72,21 +72,20 @@ When creating a dynamic menu, the structure `type` properties are used in order
 * type `submenu` must have `getSubmenuItems`, and if it has an `onAction` property it is ignored.
 * type `separator` ignores all other properties.
 
-////
-- ## Example Usage
```
[the remark this answers is not in the document]
Comment (hamza0867) [reply modules/ROOT/pages/ui-components/contextmenu.adoc]: Done

File: modules/ROOT/partials/misc/predefinedIconsOnly.adoc [hunk 4]
```diff
@@ -1 +1,2 @@
+// TODO: I wonder if these links should be replaced with xrefs
```
[the remark this answers is not in the document]
Comment (hamza0867) [reply modules/ROOT/partials/misc/predefinedIconsOnly.adoc]: Done

File: modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc [hunk 5]
```diff
@@ -259,7 +262,8 @@ A split button contains a basic button and a menu button, wrapped up into one to
 | string
 | Primary button
 | optional
-| include::partial$misc/predefinedIconsOnly.adoc[]
+|
+include::partial$misc/predefinedIconsOnly.adoc[]
 
 | onAction
 | (api) \=> void
```
[the remark this answers is not in the document]
Comment (hamza0867) [reply modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc]: Hi @lnewson, can you please mark the task above as done ? Thank you

[the remark this answers is not in the document]
Comment (lnewson) [reply modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc]: @hamza0867 it looks like there are still quite a number of issues for this in the `typeoftoolbarbuttons.adoc` file sorry.

[the remark this answers is not in the document]
Comment (lnewson) [reply modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc]: One example is `(success: (menu) \=> void) \=> void` should be this instead:
```
`(success: (menu) => void) => void`
```

[the remark this answers is not in the document]
Comment (hamza0867) [reply modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc]: Ok I see, thanks

[the remark this answers is not in the document]
Comment (lnewson) [reply modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc]: Just to clarify, you need to remove the escape chars and wrap it in backticks. I see the first part was done in 636e8c9a49fcf1296e261e9200bb826207256c36, but looks like the second part still needs doing.

[the remark this answers is not in the document]
Comment (lnewson) [reply modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc]: Looks good now, thanks!

[the remark this answers is not in the document]
Comment (hamza0867) [reply modules/ROOT/pages/ui-components/typesoftoolbarbuttons.adoc]: 👍

**Review by metricjs — APPROVED** [review]

**Pull request merged**